helpers.go

```diff
@@ -42,6 +42,34 @@ func getCmd(name string, f func(ctx *cli.Context)) func(*cli.Context) {
 	}
 }
 
+// Returns a string containing the global flags being used.
+func globalFlags() string {
+	if currentContext == nil {
+		return ""
+	}
+
+	res := ""
+	flags := []string{"non-interactive", "no-gpg-checks", "gpg-auto-import-keys"}
+
+	for _, v := range flags {
+		if currentContext.GlobalBool(v) {
+			res = res + "--" + v + " "
+		}
+	}
+	return res
+}
+
+// Concatenate the given zypper commands, while adding the global flags
+// currently in place.
+func formatZypperCommand(cmds ...string) string {
+	flags := globalFlags()
+
+	for k, v := range cmds {
+		cmds[k] = "zypper " + flags + v
+	}
+	return strings.Join(cmds, " && ")
```

Comment:
@flavio I'm not sure about this. Local command flags can be given with the `cmdWithFlags` command no ?

As an example, grab the helpers.go file from master, line 249. You can find the following:

```go
	cmd := fmt.Sprintf(
		"zypper ref && zypper -n %v",
		cmdWithFlags(zypperCmd, ctx, boolFlags, toIgnore))
```

In this case, the proper solution would be:

```go
cmd := formatZypperCommand("ref", zypperCmd)
cmd = fmt.Sprintf("-n %v", cmdWithFlags(cmd, ctx, boolFlags, toIgnore))
```

This would respect the extra `-n` global flag *and* local flags. If you wanted to be safer with "ref", you could call `formatZypperCommand` two times, once per command (and make the proper adjustments for each).